packages/xrpc/src/types.ts

```diff
@@ -188,3 +188,17 @@ export class XRPCInvalidResponseError extends XRPCError {
     )
   }
 }
+
+type IsLiteralKey<K extends PropertyKey> = {
+  string: string extends K ? 0 : 1
+  number: number extends K ? 0 : 1
+  symbol: symbol extends K ? 0 : 1
+}[K extends string ? 'string' : K extends number ? 'number' : 'symbol']
```

Comment:
This isn't 100% complete. It might not handle index signatures with template literal types or string mappings (like `Uppercase<string>`) but I seriously doubt you care about any of those here. 

In a sense, support for `number` and `symbol` is already beyond what you need here but it was easy enough to add it so I did it for the completeness' sake